daffodil-lib/src/main/scala/org/apache/daffodil/lib/util/ArrayBuffer1.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.lib.util
+
+import java.util
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Compatibility ArrayBuffer class for 2.12 and 2.13 since reduceToSize
+ * has been removed in 2.13. This allows us to maintain the same
+ * functionality as 2.12 while upgraded to 2.13
+ */
+class ArrayBuffer1[T](initialSize: Int = 1) extends ArrayBuffer[T] {
+
+  def this() = this(1) // Default size if none is provided
+
+  // Preallocate space to avoid frequent resizing
+  this.sizeHint(initialSize)
+
+  def reduceToSize1(sz: Int): Unit = {
+    if (sz >= 0 && sz <= size0) {
+      // to ensure things are not left un-garbage collected
+      util.Arrays.fill(array, sz, size0, null)
+      size0 = sz
+    } else
+      throw new IllegalArgumentException(
+        "Invalid size: Must be between 0 and the current size of the buffer."
+      )
+  }
+}
+
+object ArrayBuffer1 extends ArrayBuffer {
+  def apply[T](): ArrayBuffer1[T] = {
+    new ArrayBuffer1[T]()
+  }
+
+  def apply[A](initialSize: Int): ArrayBuffer1[A] = {
+    new ArrayBuffer1[A](initialSize)
```

Comment:
The normal object ArrayBuffer apply method creates an ArrayBuffer with those elements as the entries. It's not the size. For example, `ArrayBuffer(10)` creates an ArrayBuffer with a single element with value `10`. It also supports things like `ArrayBuffer(1, 2, 3, 4, ...)` for multiple elements. I'm not sure we have to implement this, but if we do it should proably have the same behavior as ArrayBuffer.

Reply:
We don't actually use that apply method so I removed it